core/app/models/spree/payment_method.rb

```diff
@@ -17,7 +17,10 @@ class PaymentMethod < Spree::Base
     scope :active, -> { where(active: true) }
     scope :available_to_users, -> { where(available_to_users: true) }
     scope :available_to_admin, -> { where(available_to_admin: true) }
-    scope :available_to_store, -> (store) { (store.present? && store.payment_methods.empty?) ? self : store.payment_methods }
+    scope :available_to_store, ->(store) do
+      raise ArgumentError, "You must provide a store" if store.nil?
+      store.payment_methods.empty? ? all : where(id: store.payment_method_ids)
+    end
```

Comment:
That won't quite work due to the nature of the relationship between stores and payment methods. It could be writen as:
```ruby
scope :available_to_store, -> (store) { joins(:stores).merge(Spree::Store.where(id: store.id)) }
```

However, this would mean a divergence from how we typically handle store associations. Normally, if a store doesn't have anything associated to it we assume everything belongs to that store. I believe this decision was made a while back to help handle single store apps by ensuring everything continued to work without having to create the missing associations.

While I'm definitely not opposed to changing that behavior, it would require a larger amount of work and a conscious decision to have all single store apps require some additional configuration behind the scenes.